add-on/chrome/bootstrap.js

```diff
@@ -877,17 +877,51 @@ var WindowListener = {
       _browserSharePaused: false,
 
       /**
+       * Stores details about the last notification.
+       *
+       * @type {Object}
+       */
+      _lastNotification: {},
+
+      /**
+       * Used to determine if the browser sharing info bar is currently being
+       * shown or not.
+       */
+      _showingBrowserSharingInfoBar: function() {
+        let browser = gBrowser.selectedBrowser;
+        let box = gBrowser.getNotificationBox(browser);
+        let notification = box.getNotificationWithValue(kBrowserSharingNotificationId);
+
+        return !!notification;
+      },
+
+      /**
        * Shows an infobar notification at the top of the browser window that warns
        * the user that their browser tabs are being broadcasted through the current
        * conversation.
        * @param  {String} currentRoomToken Room we are currently joined.
        * @return {void}
        */
       _maybeShowBrowserSharingInfoBar: function(currentRoomToken) {
-        this._hideBrowserSharingInfoBar();
-
         let participantsCount = this.LoopRooms.getNumParticipants(currentRoomToken);
 
+        if (this._showingBrowserSharingInfoBar()) {
+          // When we first open the room, there will be one or zero partipicants
+          // in the room. The notification box changes when there's more than one,
+          // so work that out here.
+          let participantsOneOrZero = participantsCount <= 1;
+          let lastParticipantsOneOrZero = this._lastNotification <= 1;
+
+          // If we're not actually changing the notification bar, then don't
+          // re-display it. This avoids the bar sliding in twice.
+          if (participantsOneOrZero !== lastParticipantsOneOrZero &&
+              this._lastNotification.paused === this._browserSharePaused) {
```

Comment:
nit: `this._browserSharePaused === this._lastNotification.paused` keeping the the `last` on the RHS.